packages/data-stores/src/reader/queries/use-site-subscriptions-query.ts

```diff
@@ -56,15 +68,13 @@ const useSiteSubscriptionsQuery = ( {
 
 	// Flatten all the pages into a single array containing all subscriptions
 	const flattenedData = data?.pages?.map( ( page ) => page.subscriptions ).flat();
-	// Transform the dates into Date objects
-	const transformedData = flattenedData?.map( ( subscription ) => ( {
-		...subscription,
-		last_updated: new Date( subscription.last_updated ),
-		date_subscribed: new Date( subscription.date_subscribed ),
-	} ) );
 
 	return {
```

Comment:
In case we notice some performance issues, we should keep in mind that the results here are not memoized. We iterate over all items in `flattenedData` 3 times (once above, when we initialize the const itself). I think it is good for now, though.